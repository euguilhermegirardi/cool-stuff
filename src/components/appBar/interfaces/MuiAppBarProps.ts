export interface MuiAppBarProps {
  anchorElUser: HTMLElement | null;
  drawerWidth: number;
  mobileOpen: boolean;
  pages: string[];
  settings: string[];
  handleDrawerToggle: () => void;
  handleOpenUserMenu: (event: React.MouseEvent<HTMLElement>) => void;
  handleCloseUserMenu: () => void;
  handleCloseNavMenu: () => void;
};
