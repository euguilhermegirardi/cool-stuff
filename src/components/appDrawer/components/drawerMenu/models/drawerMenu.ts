import ApplicationRoutes from 'utils/navigation/applicationRoutes';
import translations from 'utils/translations';

export interface DrawerMenuProps {
  id: number;
  path: string;
  name: string;
};

export const drawerMenu: DrawerMenuProps[] = [
  {
    id: 1,
    path: ApplicationRoutes.dashboard,
    name: translations.dashboard.title
  },
  {
    id: 2,
    path: ApplicationRoutes.pokemonList,
    name: translations.pokemonList.title
  },
  {
    id: 3,
    path: ApplicationRoutes.pokemonCard,
    name: translations.pokemonCard.title
  },
  {
    id: 4,
    path: ApplicationRoutes.pokemonStatistics,
    name: translations.pokemonStatistics.title
  },
];
