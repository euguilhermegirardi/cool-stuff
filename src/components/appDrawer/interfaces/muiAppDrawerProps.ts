export interface MuiAppDrawerProps {
  drawerWidth: number;
  mobileOpen: boolean;
  handleDrawerToggle: () => void;
};
