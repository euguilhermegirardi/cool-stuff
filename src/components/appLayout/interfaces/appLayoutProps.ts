import { ReactElement } from 'react';

export interface AppLayoutProps {
  drawerWidth: number;
  mobileOpen: boolean;
  handleDrawerToggle: () => void;
  children: ReactElement;
};
