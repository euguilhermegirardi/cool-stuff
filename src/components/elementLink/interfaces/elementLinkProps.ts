export default interface ElementLinkProps {
  link: string;
  text: string;
  onClick?: () => void;
};
