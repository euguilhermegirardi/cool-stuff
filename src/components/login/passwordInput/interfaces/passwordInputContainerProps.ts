import { UseFormRegister } from 'react-hook-form';

export interface PasswordInputContainerProps {
  autoFocus: boolean;
  register: UseFormRegister<any>;
  formErrors?: {
    message?: string;
  };
  inputId: string;
  validateInput?: () => void;
};
