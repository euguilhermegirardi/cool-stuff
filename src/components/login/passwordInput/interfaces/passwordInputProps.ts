import { UseFormRegister } from 'react-hook-form';

export interface PasswordInputProps {
  autoFocus: boolean;
  register: UseFormRegister<any>;
  formErrors?: {
    message?: string;
  };
  showPassword: boolean;
  handleShowPassword: () => void;
  handleMouseDownPassword: (
    event: React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) => void;
  inputId: string;
};
