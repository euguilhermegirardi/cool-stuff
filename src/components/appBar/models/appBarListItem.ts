import { DrawerMenuProps } from 'components/appDrawer/components/drawerMenu/models/drawerMenu';
import ApplicationRoutes from 'utils/navigation/applicationRoutes';

export interface AppBarListItems extends DrawerMenuProps { };

export const appBarListItems: AppBarListItems[] = [
  {
    id: 1,
    path: ApplicationRoutes.projectTechnologies,
    name: 'Project Technologies',
  },
  {
    id: 2,
    path: ApplicationRoutes.contactUs,
    name: 'Contact Us',
  },
];
