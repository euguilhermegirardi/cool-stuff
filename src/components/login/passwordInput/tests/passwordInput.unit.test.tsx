import { useForm } from 'react-hook-form';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MockPasswordInputProps } from './interfaces/passwordInputProps';
import { mockHandleMouseDownPassword, mockHandleShowPassword, mockInputId, mockInputName, mockPasswordEmptyFormErrors, mockPasswordFormErrors } from './models/passwordInputProps';
import PasswordInput from '../passwordInput';

describe('Password Input Component', () => {
  const PasswordInputTestContainer = ({
    inputId,
    inputName,
    showPassword,
    formErrors,
    handleMouseDownPassword,
    handleShowPassword,
  }: {
    inputId: string;
    inputName: string;
    showPassword: boolean;
    formErrors?: {
      message?: string;
    };
    handleMouseDownPassword: (
      event: React.MouseEvent<HTMLButtonElement, MouseEvent>
    ) => void;
    handleShowPassword: () => void;
  }) => {
    const { register } = useForm<MockPasswordInputProps>({
      defaultValues: {
        test: 'joe doe',
      },
    });

    return (
      <PasswordInput
        inputId={inputId}
        inputName={inputName}
        showPassword={showPassword}
        formErrors={formErrors}
        handleMouseDownPassword={handleMouseDownPassword}
        register={register}
        handleShowPassword={handleShowPassword}
      />
    );
  };

  const renderPasswordInput = (
    showPassword: boolean,
    formErrors?: {
      message?: string;
    }
  ) =>
    render(
      <PasswordInputTestContainer
        inputId={mockInputId}
        inputName={mockInputName}
        showPassword={showPassword}
        formErrors={formErrors}
        handleMouseDownPassword={mockHandleMouseDownPassword}
        handleShowPassword={mockHandleShowPassword}
      />
    );

  it('should render the component', () => {
    renderPasswordInput(false, mockPasswordEmptyFormErrors);
    expect(screen).not.toBeNull();
  });

  it('should render the input with the image icon', () => {
    renderPasswordInput(false, mockPasswordEmptyFormErrors);
    const imgLocker = screen.getByRole('img', { name: /password\-locker/i });
    const passwordInput = screen.getByPlaceholderText(/password/i);
    const showPasswordBtn = screen.getByRole('button', {
      name: /toggle password visibility/i
    });
    const visibilityOffIcon = screen.getByTestId('VisibilityOffIcon');

    expect(imgLocker).toBeInTheDocument();
    expect(passwordInput).toBeInTheDocument();
    expect(showPasswordBtn).toBeInTheDocument();
    expect(visibilityOffIcon).toBeInTheDocument();
  });

  it('should be able to type in the input with the visibilityOnIcon', async () => {
    renderPasswordInput(true, mockPasswordEmptyFormErrors);
    const passwordInput = screen.getByPlaceholderText(/password/i);
    const visibilityIcon = screen.getByTestId('VisibilityIcon');

    expect(passwordInput).toHaveValue('');
    await userEvent.type(passwordInput, 'testing the password input');
    expect(passwordInput).toHaveValue('testing the password input');
    expect(visibilityIcon).toBeInTheDocument();
  });

  it('should render the form error message', () => {
    renderPasswordInput(false, mockPasswordFormErrors);
    const errorMessage = screen.getByText(mockPasswordFormErrors.message);

    expect(errorMessage).toBeInTheDocument();
  });

  it('should match the snapshot with form errors', () => {
    const passwordInputComponent = renderPasswordInput(false, mockPasswordFormErrors);
    expect(passwordInputComponent).toMatchSnapshot();
  });

  it('should match the snapshot without form errors', () => {
    const passwordInputComponent = renderPasswordInput(false, mockPasswordEmptyFormErrors);
    expect(passwordInputComponent).toMatchSnapshot();
  });
});
