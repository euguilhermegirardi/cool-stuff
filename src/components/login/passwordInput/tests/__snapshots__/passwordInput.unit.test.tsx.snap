// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Password Input Component > should match the snapshot with form errors 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-2ehmn7-MuiInputBase-root-MuiOutlinedInput-root"
          style="height: 40px; border-radius: 0; margin-bottom: 8px;"
        >
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-vpsq89-MuiInputAdornment-root"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
            <img
              alt="password-locker"
              class="MuiBox-root css-10lx7i0"
              src="/src/assets/icons/locker.svg"
            />
          </div>
          <input
            aria-invalid="true"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-152mnda-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r4:"
            name="password"
            placeholder="PasswordInputTest"
            type="password"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="toggle password visibility"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1us4s92-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="VisibilityOffIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-180x7my-MuiTypography-root"
        role="alert"
      >
        password is required
      </p>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-2ehmn7-MuiInputBase-root-MuiOutlinedInput-root"
        style="height: 40px; border-radius: 0; margin-bottom: 8px;"
      >
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-vpsq89-MuiInputAdornment-root"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
          <img
            alt="password-locker"
            class="MuiBox-root css-10lx7i0"
            src="/src/assets/icons/locker.svg"
          />
        </div>
        <input
          aria-invalid="true"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-152mnda-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r4:"
          name="password"
          placeholder="PasswordInputTest"
          type="password"
          value=""
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
        >
          <button
            aria-label="toggle password visibility"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1us4s92-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="VisibilityOffIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-180x7my-MuiTypography-root"
      role="alert"
    >
      password is required
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Password Input Component > should match the snapshot without form errors 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-2ehmn7-MuiInputBase-root-MuiOutlinedInput-root"
          style="height: 40px; border-radius: 0; margin-bottom: 8px;"
        >
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-vpsq89-MuiInputAdornment-root"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
            <img
              alt="password-locker"
              class="MuiBox-root css-10lx7i0"
              src="/src/assets/icons/locker.svg"
            />
          </div>
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-152mnda-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r5:"
            name="password"
            placeholder="PasswordInputTest"
            type="password"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="toggle password visibility"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1us4s92-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="VisibilityOffIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-2ehmn7-MuiInputBase-root-MuiOutlinedInput-root"
        style="height: 40px; border-radius: 0; margin-bottom: 8px;"
      >
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-vpsq89-MuiInputAdornment-root"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
          <img
            alt="password-locker"
            class="MuiBox-root css-10lx7i0"
            src="/src/assets/icons/locker.svg"
          />
        </div>
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-152mnda-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r5:"
          name="password"
          placeholder="PasswordInputTest"
          type="password"
          value=""
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
        >
          <button
            aria-label="toggle password visibility"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1us4s92-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="VisibilityOffIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
